### PR TITLE
support for modules and interfaces

### PR DESCRIPTION
**Pull request by tineheller**

### CODE REVIEW

File: src/core/jsdoc-tsd-parser.ts [hunk 1]
```diff
@@ -350,8 +378,87 @@ export class JSDocTsdParser {
...
+								}
+
+								(parentItem as dom.InterfaceDeclaration).members.push(objectTypeMember);
+								break;
+
+							case "module":
+								let moduleMember = parsedItem as dom.ModuleMember;
+
+								switch ((moduleMember as any).kind) {
+									case "member":
```
Comment (wehrstedt) [inline src/core/jsdoc-tsd-parser.ts]: You make every module member to a constant member which is not correct. A non-constant module member would be from type "var". This is currently not supported by the dts-dom module so we have to create a pull request to make this type of module members available.
For now you can either remove these cases or add the ability to parse constants. Then you have to fix this cases that only constants will be exported to constants

File: src/core/jsdoc-tsd-parser.ts [hunk 2]
```diff
@@ -350,8 +378,87 @@ export class JSDocTsdParser {
 								}
 								break;
 
+							case "interface":
+								let objectTypeMember = parsedItem as dom.ObjectTypeMember;
+
+								switch ((objectTypeMember as any).kind) {
+									case "function":
+										let functionDeclaration: any = objectTypeMember as any;
+										objectTypeMember = {
```
Comment (wehrstedt) [inline src/core/jsdoc-tsd-parser.ts]: Maybe we should create the method declaration with the dts-dom-module like dom.create.method(...). It's more safe if the API changes in future..

File: src/core/jsdoc-tsd-parser.ts [hunk 3]
```diff
@@ -266,6 +293,7 @@ export class JSDocTsdParser {
 		this.resultItems[jsdocItem.longname].push(domInterface);
 	}
 
+	// tslint:disable-next-line:member-ordering
```
Comment (wehrstedt) [inline src/core/jsdoc-tsd-parser.ts]: can you fix the member ordering instead of ignore this error?

File: src/core/jsdoc-tsd-parser.ts [hunk 4]
```diff
@@ -236,6 +256,13 @@ export class JSDocTsdParser {
 		}
 	}
 
+	private parseModule(jsdocItem: INamespaceDoclet) {
+		let domModule: dom.ModuleDeclaration = dom.create.module(jsdocItem.name);
+		domModule.jsDocComment = this.cleanJSDocComment(jsdocItem.description);
```
Comment (wehrstedt) [inline src/core/jsdoc-tsd-parser.ts]: Take a look at #9

File: src/core/jsdoc-tsd-parser.ts [hunk 5]
```diff
@@ -104,7 +117,7 @@ export class JSDocTsdParser {
 					.trim();
 
 				// ignore everything that is not part of the function description in tsd-files
-				if (cleanedLine && (cleanedLine.startsWith("@param") || cleanedLine.startsWith("@throws") || !cleanedLine.startsWith("@"))) {
+				if (cleanedLine && (cleanedLine.startsWith("@param") || cleanedLine.startsWith("@throws") || cleanedLine.startsWith("@description") || !cleanedLine.startsWith("@"))) {
```
Comment (wehrstedt) [inline src/core/jsdoc-tsd-parser.ts]: You don't have to enable the @description-Tag. Your parsed `jsdocItem` contains an property 'description' which contains only the value which is annotated with "@description". In addition the annotation you can omit this annotation by adding your description on top of the jsdoc comment without any annotations.

File: src/core/jsdoc-tsd-parser.ts [hunk 6]
```diff
@@ -91,6 +103,7 @@ export class JSDocTsdParser {
 		return domParams;
 	}
 
+	// tslint:disable-next-line:member-ordering
```
Comment (wehrstedt) [inline src/core/jsdoc-tsd-parser.ts]: Maybe we should create the property declaration with the dts-dom-module like dom.create.property(...). It's more safe if the API changes in future..

**Comment by tineheller** [comment]
Actually I think you've requested a lot of changes. Also we still have to wait for the changes for variables in modules in the dts-dom module. So I think it's best, if I create new smaller pull requests and close this one. Then it'll be easier to keep track of all the changes. And I cann add interfaces and change the description first and add the modules later.

**Comment by wehrstedt** [comment]
If it's easier for you we can make multiple issues so that you can split these changes into multiple, smaller pull requests. I will create some issues.

**Pull request closed without merging**